### PR TITLE
Colorspace CieXyz

### PR DESCRIPTION
**Pull request by dampee**
A "copy-paste" of the methods mentioned in issue #260 
(https://github.com/scijs/color-space/blob/master/xyz.js)

### CODE REVIEW

**Comment by dampee** [comment]
The calcuations do *not* match 100% with [mathLab documentation examples](http://nl.mathworks.com/help/images/ref/xyz2rgb.html).
You'll see in the unittests that the precision is altered to make them work.

File: src/ImageProcessor/Colors/Colorspaces/CieXyz.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        /// An instance of <see cref="CieXyz"/>.
+        /// </returns>
+        public static implicit operator CieXyz(Color color)
+        {
+            var r = color.R;
+            var g = color.G;
+            var b = color.B;
+
+            // assume sRGB
+            r = r > 0.04045 ? (float)Math.Pow(((r + 0.055) / 1.055), 2.4) : (float)(r / 12.92);
```
Comment (JimBobSquarePants) [inline src/ImageProcessor/Colors/Colorspaces/CieXyz.cs]: Got you covered already. Use `Color.InverseCompand(color)` here. :smile:

File: src/ImageProcessor/Colors/ColorTransforms.cs [hunk 2]
```diff
@@ -67,6 +67,31 @@ public partial struct Color
...
+            float g = (x * -0.96924363628087F) + (y * 1.87596750150772F) + (z * 0.041555057407175F);
+            float b = (x * 0.055630079696993F) + (y * -0.20397695888897F) + (z * 1.056971514242878F);
+
+            r = r > 0.0031308 ? (float)((1.055 * Math.Pow(r, 1.0 / 2.4)) - 0.055) : r = (r * 12.92F);
+
+            g = g > 0.0031308 ? (float)((1.055 * Math.Pow(g, 1.0 / 2.4)) - 0.055) : g = (g * 12.92F);
+
+            b = b > 0.0031308 ? (float)((1.055 * Math.Pow(b, 1.0 / 2.4)) - 0.055) : b = (b * 12.92F);
+
+            r = Math.Min(Math.Max(0, r), 1);
```
Comment (JimBobSquarePants) [inline src/ImageProcessor/Colors/ColorTransforms.cs]: Got you covered already. 
Use `Color.Compand(color)`
Return `color.Limited`. :smile:

File: src/ImageProcessor/Colors/Colorspaces/CieXyz.cs [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            g = g > 0.04045 ? (float)Math.Pow(((g + 0.055) / 1.055), 2.4) : (float)(g / 12.92);
+            b = b > 0.04045 ? (float)Math.Pow(((b + 0.055) / 1.055), 2.4) : (float)(b / 12.92);
+
+            var x = (r * 0.41239079926595F) + (g * 0.35758433938387F) + (b * 0.18048078840183F);
```
Comment (JimBobSquarePants) [inline src/ImageProcessor/Colors/Colorspaces/CieXyz.cs]: Could we use explicit type declaration here. I find var can sometimes lead to subtle numerical errors that can be hard to debug.

**Comment by JimBobSquarePants** [comment]
This is excellent. Thank you! :+1: 

I've made a few line notes that'll clean things up a little. We should definitely test thoroughly against ColorMine. I use [ColorHexa](http://www.colorhexa.com/) to compare against also.

**Pull request merged**